src/command_modules/azure-cli-interactive/azclishell/telemetry.py

```diff
@@ -19,27 +22,40 @@
 INSTRUMENTATION_KEY = '762871d5-45a2-4d67-bf47-e396caf53d9d'
 
 
-def my_context(tel_client):
-    """ context for the application """
-    tel_client.context.application.id = 'Azure CLI Shell'
-    tel_client.context.application.ver = __version__
-    tel_client.context.user.id = Profile().get_installation_id()
-    tel_client.context.instrumentation_key = INSTRUMENTATION_KEY
+def set_custom_properties(prop, name, value):
+    actual_value = value() if hasattr(value, '__call__') else value
+    if actual_value:
+        prop['az/interactive/' + name] = actual_value
```

Comment:
Property name should follow this prefix: https://github.com/Azure/azure-cli/blob/master/src/azure-cli-core/azure/cli/core/telemetry.py#L22